src/classes/CRLP_Batch_Base_NonSkew.cls

```diff
@@ -36,6 +36,13 @@
 */
 public abstract class CRLP_Batch_Base_NonSkew extends CRLP_Batch_Base {
 
+    /*******************************************************************************************************
+    * @description When Chunking mode is used for an LDV org, this is the last Id retrieved
+    * in the previous batch. When the job is chained, it'll query records with Ids greater than this Id.
+    */
+    @TestVisible
+    private Id chunkModeLastProcessedId;
```

Comment:
What would you think of `lastIdProcessed` for this, and then not check for `isChunkEnabled` on line 151 before assigning to this var?

Reply:
I like the name change - updated!  We need the `isChunkModeEnabled`  check on 145 because in the first `execute` method in the first batch job, `lastProcessedId` will be null here, so in that specific context we can only know if we are in chunk mode by checking that the custom setting value is configured (and if its not configured than we don't need to store the Id).